Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -0,0 +1,1094 @@
+        c4Params.collections = replCollections.data();
+    };
+    replicate(paramsSetter);
+    // total 3 docs, 12 revs.
+    CHECK(_callbackStatus.progress.documentCount == 12);
+    verifyDocs(collectionSpecs, docIDs, true);
+}
+
+#if 0
+TEST_CASE_METHOD(ReplicatorCollectionTest, "Reset Checkpoint with Pull", "[Pull]") {
+    addDocs(db, Roses, 10);
+    addDocs(db, Tulips, 10);
+    
+    _expectedDocumentCount = 20;
+    runPullReplication({Tulips, Lavenders, Roses}, {Roses, Tulips});
+    validateCollectionCheckpoints(db2, db, 0, "{\"remote\":10}");
+    validateCollectionCheckpoints(db2, db, 1, "{\"remote\":10}");
+    
+    purgeAllDocs(db2, Roses);
+    purgeAllDocs(db2, Tulips);
+    
+    _expectedDocumentCount = 0;
+    runPullReplication({Tulips, Lavenders, Roses}, {Roses, Tulips});
+    validateCollectionCheckpoints(db2, db, 0, "{\"remote\":10}");
+    validateCollectionCheckpoints(db2, db, 1, "{\"remote\":10}");
+    
+    _expectedDocumentCount = 20;
+    runPullReplication({Tulips, Lavenders, Roses}, {Roses, Tulips}, kC4OneShot, true);
+    validateCollectionCheckpoints(db2, db, 0, "{\"remote\":10}");
+    validateCollectionCheckpoints(db2, db, 1, "{\"remote\":10}");
+}
+#endif
+
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Push and Pull Attachments SG", "[.SyncServerCollection]") {
+    string idPrefix = timePrefix();
+    //    constexpr size_t collectionCount = 2;
+    constexpr size_t collectionCount = 1;
+    std::array<C4CollectionSpec, collectionCount> collectionSpecs = {
+        Roses
+        //, Tulips
+    };
+    std::array<C4Collection*, collectionCount> collections =
+        collectionPreamble(collectionSpecs, "sguser", "password", kC4LogNone);
+    std::array<unordered_map<alloc_slice, unsigned>, collectionCount> docIDs;
+    std::array<C4ReplicationCollection, collectionCount> replCollections;
+    vector<string> attachments1 = {idPrefix+"Attachment A", idPrefix+"Attachment B", idPrefix+"Attachment Z"};
+    std::array<vector<C4BlobKey>, collectionCount> blobKeys; // blobKeys1a, blobKeys1b;
+    {
+        string doc1 = idPrefix + "doc1";
+        string doc2 = idPrefix + "doc2";
+        TransactionHelper t(db);
+        for (size_t i = 0; i < collectionCount; ++i) {
+            blobKeys[i] = addDocWithAttachments(db, collectionSpecs[i],  slice(doc1), attachments1, "text/plain");
+            docIDs[i] = getDocIDs(collections[i]);
+            replCollections[i] = C4ReplicationCollection{collectionSpecs[i], kC4OneShot, kC4Disabled};
+        }
+    }
+
+    C4ParamsSetter paramsSetter = [&replCollections](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = replCollections.size();
+        c4Params.collections = replCollections.data();
+    };
+    replicate(paramsSetter);
+    verifyDocs(collectionSpecs, docIDs);
+    for (size_t i = 0; i < collectionCount; ++i) {
+        checkAttachments(verifyDb, blobKeys[i], attachments1);
+    }
+}
+
+#if 0
```

Comment:
Is this test failed? Can you give info and ticket number if having one about the reason to disable this test?